### DISCUSSION TITLE
Zeek not generating intel.log

### DISCUSSION THREAD

**Discussion by mdavis332** [Security-Onion-Solutions/securityonion#5549] [Unsupported Versions]
I've been unable to get an `intel.log` to generate from Zeek Intel since I setup SO several versions ago. I keep hoping on each version upgrade that it'll magically work but no joy so far. I'm currently on the latest version as of this post (2.3.70) confirmed by `sudo soup`.

I've read and followed the directions here: https://docs.securityonion.net/en/2.3/zeek.html#intel

My `/opt/so/saltstack/local/salt/zeek/policy/intel/intel.dat` file looks like the following:
![intel dat](https://user-images.githubusercontent.com/1747636/133693019-7033b71a-5fba-436b-9b80-9298ef438157.jpg)

If I `cat` my `/nsm/zeek/logs/current/loaded_scripts.log` it ends with this:
![loaded_scripts](https://user-images.githubusercontent.com/1747636/133693218-6f322a81-f3e9-44a8-ba44-1ca7f0806be2.jpg)

If I `sudo docker exec -it <id> bash` into the so-zeek container, I've verified that `so-zeek:/opt/zeek/share/zeek/policy/intel/intel.dat` has the right contents.  No `/nsm/zeek/logs/current/reporter.log` is ever generated with any txt relevant to this issue.

Not sure what else to look for atm. Been playing with it off and on for a few months and would like to get it working. At the risk of hijacking my own thread, I'm also not getting ja3 hashes (ja3s shows up in ssl.log, but never ja3). I may open a separate thread on that one...

**Comment by dougburks** [comment]
Have you verified that Zeek is actually seeing traffic to reddit.com?

Have you double-check the formatting of your intel.dat to ensure proper tabbing and no extra spaces?

**Comment by mdavis332** [comment]
Howdy Doug,

Thanks for the suggestions. 

> Have you verified that Zeek is actually seeing traffic to reddit.com?

I just double-checked again and, indeed, `reddit.com` immediately shows up in `dns.log` as a query with expected answers and the `id.orig_h` field correctly displaying the host on which reddit was browsed. So Zeek does appear to be seeing the DNS lookup.

> Have you double-check the formatting of your intel.dat to ensure proper tabbing and no extra spaces?

Here's a vim of my `intel.dat` with vim's `:set list` enabled to show characters like tabs (^I)

![image](https://user-images.githubusercontent.com/1747636/134024419-3382ec69-2713-47d9-9f93-09854d525fa1.png)

and here's the same thing with `:set nolist`

![image](https://user-images.githubusercontent.com/1747636/134024572-ac6d82a5-2968-49ed-b082-618a348f081e.png)

I _think_ that looks right, but if you spot something awry I'm definitely keen to fix.

Thanks again,
Michael

**Reply by dougburks** [reply]
Have you verified that there are no blank lines in the file?

**Reply by mdavis332** [reply]
D'oh, I cut off the bottom of the screenshot which makes it impossible to tell =/ but no, there's nothing after line 2. I've read that Zeek can be persnickety on the formatting/whitespacing of the `intel.dat` file, so I've tried to ensure that it follows the conventions I've read about.

I'm getting lots of other Zeek logs in the `/nsm/zeek/logs/current` dir including conn, dns, files, http, notice, ntp, ssl, etc. Just never anything for intel :(

**Reply by dougburks** [reply]
Yes, Zeek can definitely be persnickety on the formatting and whitespace of the intel.dat file! I know that it's working properly on my 2.3.70 deployments. You might try starting over with a brand new intel.dat file or creating a new installation (perhaps in a VM) to test side by side.

**Reply by mdavis332** [reply]
Understood. I sure wish Zeek had some further debug logging that could reveal what it's choosing to do (or not) and why. `reporter.log` is the only thing I've read that holds any of that type of info and there's nothing in there related to the `intel.log` in my case.
